src/context/theme-context.js

```diff
@@ -0,0 +1,42 @@
+import { useState, createContext, useContext, useEffect } from "react"
+
+const ThemeContext = createContext();
+
+const ThemeProvider = ({ children }) => {
+    const [theme, setTheme] = useState();
+
+    const toggleTheme = () => {
+        if (localStorage.getItem("theme") === "light") {
+            darkThemeBtn();
+        }
+        else {
+            lightThemeBtn();
+        }
+    }
+    const darkThemeBtn = () => {
+        document.body.classList.add("dark-theme");
+        localStorage.setItem("theme", "dark");
+        setTheme("dark");
+    }
+    const lightThemeBtn = () => {
+        document.body.classList.remove("dark-theme");
+        localStorage.setItem("theme", "light");
+        setTheme("light")
+    }
```

Comment:
These functions have same logic so you can create single function as pass the variable theme as a parameter

Reply:
sure